cmake/app/boilerplate.cmake

```diff
@@ -77,8 +77,12 @@ set(PROJECT_SOURCE_DIR $ENV{ZEPHYR_BASE})
 # Convert path to use the '/' separator
 string(REPLACE "\\" "/" PROJECT_SOURCE_DIR ${PROJECT_SOURCE_DIR})
 
+# Remove trailing '/', it results in ugly paths and also exposes some bugs
+string(REGEX REPLACE "\/+$" "" PROJECT_SOURCE_DIR ${PROJECT_SOURCE_DIR})
+
 set(ZEPHYR_BINARY_DIR ${PROJECT_BINARY_DIR})
 set(ZEPHYR_BASE ${PROJECT_SOURCE_DIR})
+set(ENV{ZEPHYR_BASE}   ${ZEPHYR_BASE})
```

Comment:
We also need to add

ZEPHYR_BASE=${ZEPHYR_BASE}

in kconfig.cmake for custom targets.

Otherwise menuconfig would be using the unfixed env var ZEPHYR_BASE.

Reply:
Forgot to add the file I'll add it in a sec.